exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -49,6 +49,7 @@
   String USER_RPC_TIMEOUT = "drill.exec.rpc.user.timeout";
   String METRICS_CONTEXT_NAME = "drill.exec.metrics.context";
   String USE_IP_ADDRESS = "drill.exec.rpc.use.ip";
+  String BIT_ADVERTISED_HOST = "drill.exec.rpc.bit.advertised.host";
```

Comment:
That all seems reasonable to me.  I can't find any examples of environment variables being accessed from in the Java code.  I assume we don't want the name of the environment variable hard-coded in the hostname resolution code, but I'm not sure of the best place to define it, i.e. ```String DRILL_HOST_NAME = "DRILL_HOST_NAME"```
Thoughts?  I could put it in `ExecConstants` but it would be the only env variable in there.  I could also just make it a static variable in the `ServiceEngine` class.

Do we still want to provide the option to override through a system property, or just rely on the environment variable?